Compiler/src/Error.hs

```diff
@@ -23,15 +54,22 @@ instance PrettyPrint Error where
     prettyPrint (LookupError (LookupErrorType s)) = ["LookupError: Type " ++ s ++ " is not defined"]
     prettyPrint (LookupError (LookupErrorTypeDecl (TransformerDecl tx _ _ _))) = ["LookupError: expected type but got transformer" ++ show tx]
 
--- dummy values that are returned as proxies when errors are encountered
-dummyBaseType :: BaseType
-dummyBaseType = Bot
-
-dummyType :: Type
-dummyType = (Any, Bot)
+groupErrors :: [ErrorCat] -> ([Error Preprocessing], [Error Typechecking], [Error Compiling])
+groupErrors [] = ([], [], [])
+groupErrors (x:xs) = (groupParsed ++ packedParsed, groupPreprocessed ++ packedPreprocessed, groupTypechecked ++ packedTypechecked)
+    where
+        (packedParsed, packedPreprocessed, packedTypechecked) = case x of
+            (PreprocessorError e) -> ([e], [], [])
+            (TypecheckerError e) -> ([], [e], [])
+            (CompilerError e) -> ([], [], [e])
+        (groupParsed, groupPreprocessed, groupTypechecked) = groupErrors xs
 
-dummyDecl :: Decl
-dummyDecl = TypeDecl "unknownDecl__" [] dummyBaseType
+indentString :: String -> String
+indentString = unlines . map indent . lines
 
-dummySolExpr :: SolExpr
-dummySolExpr = SolVar "unknownExpr__"
+instance PrettyPrint [ErrorCat] where
+    prettyPrint errors = filter (not . null) [ if (not . null) preprocessorErrors then "Preprocessor errors\n" ++ indentString (intercalate "\n" (map prettyStr preprocessorErrors)) else "",
```

Comment:
`prettyPrint` is supposed to return a list of Strings, each string representing a line, so having there be `\n` in the strings is not ideal; I think we can easily restore this property by just changing it to be something like 
```
(if (not . null) preprocessorErrors 
then "Preprocessor errors" : map indent (concatMap prettyPrint preprocessorErrors)) 
else []) ++ --- The rest of the error groups
```